### PR TITLE
[runtime-security] add trace/span context to event

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR introduces span/trace context to events. It add eRPC operation handling TLS registration. The span/trace context is read from the TLS memory and added to each event.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe how to test your changes

Write here in detail how you have tested your changes
and instructions on how this should be tested in QA.

Describe or link instructions to set up environment 
for testing, if the process requires more than just
running the agent on one of the supported platforms.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [ ] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

**Review by lebauce — APPROVED** [review]

File: pkg/security/ebpf/c/process.h [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <linux/sched.h>
 
 #include "container.h"
+#include "span.h"
```
Comment (Gui774ume) [inline pkg/security/ebpf/c/process.h:8]: I don't think this is needed

File: pkg/security/ebpf/c/exec.h [hunk 2]
```diff
@@ -392,8 +395,9 @@ int kprobe_do_exit(struct pt_regs *ctx) {
         struct exit_event_t event = {};
         struct proc_cache_t *cache_entry = fill_process_context(&event.process);
         fill_container_context(cache_entry, &event.container);
-
         send_event(ctx, EVENT_EXIT, event);
+
```
Comment (Gui774ume) [inline pkg/security/ebpf/c/exec.h:399]: I think you're missing the `fill_span_context()` here, and another one is missing in `kprobe_security_bprm_committed_creds` and in `sched_process_fork` but I can't comment there :)

File: pkg/security/ebpf/c/erpc.h [hunk 3]
```diff
@@ -43,38 +45,38 @@ struct discard_pid_t {
...
+        return 0;
+    }
+
     struct discard_pid_t discarder;
     bpf_probe_read(&discarder, sizeof(discarder), data);
 
     return discard_pid(discarder.req.event_type, discarder.pid, discarder.req.timeout);
 }
 
 int __attribute__((always_inline)) is_erpc_request(struct pt_regs *ctx) {
```
Comment (Gui774ume) [inline pkg/security/ebpf/c/erpc.h:79]: We should remove `erpc_fd` in user space too :)

**Review by Gui774ume — COMMENTED** [review]
Looks great !

File: pkg/security/tests/span_test.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	sprobe "github.com/DataDog/datadog-agent/pkg/security/probe"
+	"github.com/DataDog/datadog-agent/pkg/security/rules"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSpan(t *testing.T) {
+	ruleDef := &rules.RuleDefinition{
+		ID:         "test_span_rule",
+		Expression: `open.file.path == "{{.Root}}/test-span"`,
+	}
```
Comment (Gui774ume) [inline pkg/security/tests/span_test.go:26]: Should we test exec events too ?

Comment (safchain) [reply pkg/security/tests/span_test.go:28]: It is not possible  :(, as it requires to control the thread-id

Comment (Gui774ume) [reply pkg/security/tests/span_test.go:28]: 👍🏻

File: pkg/security/tests/schemas/schemas.go [hunk 5]
```diff
@@ -0,0 +1,829 @@
+// Code generated by go-bindata. DO NOT EDIT.
```
Comment (paulcacheux) [inline pkg/security/tests/schemas/schemas.go:1]: The schema files are included in the schema embed now

**Review by sarina-dd — APPROVED** [review]

**Pull request merged**